File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Stack/Valid_Parentheses/valid_parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> stack = new Stack<char>();
        
        foreach (char c in s) {
            if (c == '(') {
                stack.Push(')');
            } else if (c == '{') {
                stack.Push('}');
            } else if (c == '[') {
                stack.Push(']');
            } else {
                // If stack is empty or top doesn't match
                if (stack.Count == 0 || stack.Pop() != c) {
                    return false;
                }
            }
        }
        
        return stack.Count == 0;
    }
}

File: hushh-adk-agents/._kushal_profile_data/Interview-Prep/Problems_By_Pattern/Stack/Valid_Parentheses/valid_parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    // Stack to keep track of opening brackets
    let stack = [];
    
    // Map for easy lookup of matching pairs
    const map = {
        ')': '(',
        '}': '{',
        ']': '['
    };

    for (let i = 0; i < s.length; i++) {
        const char = s[i];
        
        // If it's a closing bracket
        if (map[char]) {
            // Pop the top element (or use dummy value if empty)
            const topElement = stack.length === 0 ? '#' : stack.pop();
            
            // If the popped element doesn't match the corresponding opening bracket
            if (topElement !== map[char]) {
                return false;
            }
        } else {
            // It's an opening bracket, push to stack
            stack.push(char);
        }
    }

    // If stack is empty, all brackets were matched
    return stack.length === 0;
};
